src/Domain/CampaignRepository.php

```diff
@@ -335,28 +338,46 @@ public function totalAmountRaised(int $campaignId): Money
 
         Assertion::numeric($donationSum);
 
-        $matchedFundQuery = $this->getEntityManager()->createQuery(
+        if ($matchFundsUsed === null) {
+            $matchFundsUsed = $this->totalMatchFundsUsed($campaignId);
+        }
+
+        $total = \bcadd($donationSum, $matchFundsUsed->toNumericString(), 2);
+
+        $currency = Currency::fromIsoCode($donationResult[0]['currencyCode']);
+
+        return Money::fromNumericString($total, $currency);
```

Comment:
Can make this slightly shorter (and DRY up '2' and use of bcadd, which should increase the mutation coverage metric) by converting to money object before adding:


```suggestion
        $donationSum = Money::fromNumericString($donationSum, Currency::fromIsoCode($donationResult[0]['currencyCode']));
        
        return $donationSum->plus($matchFundsUsed);
```

Reply:
I realise that the use of bcadd here is just mobed from elsewhere in the file. I think it's probably worth trying to keep money arithmetic within the money class.

Reply:
Thanks, done in 6e5f2e022c73d026c1a8d6c670a1340a287d75f8

Reply:
6e5f2e022c73d026c1a8d6c670a1340a287d75f8 LGTM.